### PR TITLE
Add extension mechanism for export other feature

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

Capture

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-query-builder/style/_query-builder.scss [hunk 1]
```diff
@@ -979,3 +979,93 @@
     padding: 2rem;
   }
 }
+
+.query-builder__export-other-viewer {
+  &__modal {
+    padding: 0;
+    height: 30vh;
```
Comment (xannem) [inline packages/legend-query-builder/style/_query-builder.scss:986]: Perhaps add increased sizing for height so codeeditor has more space for content?

File: .changeset/ninety-horses-fry.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-query-builder': patch
+---
+
+Add extension mechanism for export other feature.
```
Comment (akphi) [inline .changeset/ninety-horses-fry.md:5]:
```suggestion
Add extension mechanism for export query to other formats.
```

File: .changeset/ninety-horses-fry.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-query-builder': patch
```
Comment (akphi) [inline .changeset/ninety-horses-fry.md:2]:
```suggestion
'@finos/legend-query-builder': minor
```

File: packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts [hunk 4]
```diff
@@ -49,4 +56,9 @@ export interface QueryBuilder_LegendApplicationPlugin_Extension
    * Get the list of dataset entitlement access report action configurations.
    */
   getExtraDatasetEntitlementAccessNotGrantedReportActionConfigurations?(): DatasetEntitlementAccessReportActionConfiguration[];
+
+  /**
+   * Get the list of export other options
+   */
+  getExportOtherOptions?(): ExportOtherOption[];
```
Comment (akphi) [inline packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts:63]:
```suggestion
  getExtraQueryUsageConfigurations?(): QueryUsageConfiguration[];
```

File: packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts [hunk 5]
```diff
@@ -38,6 +38,13 @@ export type DatasetEntitlementAccessReportActionConfiguration = {
   ) => React.ReactNode;
 };
 
+export type ExportOtherOption = {
+  key: string;
+  title: string;
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts:43]: @xannem not sure I get your point here

File: packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts [hunk 6]
```diff
@@ -38,6 +38,13 @@ export type DatasetEntitlementAccessReportActionConfiguration = {
...
+  icon?: React.ReactNode | undefined;
+  content: string;
```
Comment (akphi) [inline packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts:45]: I foresee we would need to have more than just the text here, this should be a renderer for the tab content, think about something similar to `DataSpace` quick start

File: packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts [hunk 7]
```diff
@@ -49,4 +56,9 @@ export interface QueryBuilder_LegendApplicationPlugin_Extension
    * Get the list of dataset entitlement access report action configurations.
    */
   getExtraDatasetEntitlementAccessNotGrantedReportActionConfigurations?(): DatasetEntitlementAccessReportActionConfiguration[];
+
+  /**
+   * Get the list of export other options
```
Comment (akphi) [inline packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts:61]:
```suggestion
   * Get the list of query usage configurations
```

File: packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts [hunk 8]
```diff
@@ -38,6 +38,13 @@ export type DatasetEntitlementAccessReportActionConfiguration = {
   ) => React.ReactNode;
 };
 
+export type ExportOtherOption = {
```
Comment (akphi) [inline packages/legend-query-builder/src/stores/QueryBuilder_LegendApplicationPlugin_Extension.ts:41]:
```suggestion
export type QueryExportUsageConfiguration = {
```

File: packages/legend-query-builder/src/components/QueryBuilderExportOtherViewer.tsx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+  CODE_EDITOR_LANGUAGE,
+  CODE_EDITOR_THEME,
+} from '@finos/legend-lego/code-editor';
+import { useApplicationStore } from '@finos/legend-application';
+import type { QueryBuilder_LegendApplicationPlugin_Extension } from '../stores/QueryBuilder_LegendApplicationPlugin_Extension.js';
+import { useState } from 'react';
+import { guaranteeNonNullable } from '@finos/legend-shared';
+import type { QueryBuilderResultState } from '../stores/QueryBuilderResultState.js';
+
+export const QueryBuilderExportOtherViewer = observer(
```
Comment (akphi) [inline packages/legend-query-builder/src/components/QueryBuilderExportOtherViewer.tsx:40]: Let's rename all `ExportOther` -> `Usage`, e.g. here `QueryUsageViewer`

File: packages/legend-query-builder/src/stores/QueryBuilderResultState.ts [hunk 10]
```diff
@@ -81,6 +81,7 @@ export class QueryBuilderResultState {
   executionDuration?: number | undefined;
   latestRunHashCode?: string | undefined;
   queryRunPromise: Promise<ExecutionResult> | undefined = undefined;
+  isExportOtherModalOpened = false;
```
Comment (akphi) [inline packages/legend-query-builder/src/stores/QueryBuilderResultState.ts:84]: let's rename these and its related to `isQueryUsageViewerOpen`

File: packages/legend-query-builder/src/components/QueryBuilderResultPanel.tsx [hunk 11]
```diff
@@ -1041,6 +1042,15 @@ export const QueryBuilderResultPanel = observer(
...
                     </MenuContentItem>
                   ))}
+                  <MenuContentItem
+                    className="query-builder__result__export__dropdown__menu__item"
+                    onClick={(): void =>
+                      resultState.setIsExportOtherModalOpened(true)
+                    }
+                    disabled={queryBuilderState.changeDetectionState.hasChanged}
+                  >
+                    Other
```
Comment (akphi) [inline packages/legend-query-builder/src/components/QueryBuilderResultPanel.tsx:1052]: Suffix `...` as this prompts user for further actions

```suggestion
                    View Query Usage...
```

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-query-builder/src/components/QueryUsageViewer.tsx [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+        open={resultState.isQueryUsageViewerOpened}
+        onClose={() => resultState.setIsQueryUsageViewerOpened(false)}
+      >
+        <Modal className="query-builder__usage-viewer__modal" darkMode={true}>
+          <ModalHeader title="Query Usage Options" />
+          <ModalBody className="query-builder__usage-viewer__body">
+            <>
+              {extraQueryUsageOptions.length === 0 ? (
+                <BlankPanelContent>
+                  Export options are not available
```
Comment (akphi) [inline packages/legend-query-builder/src/components/QueryUsageViewer.tsx:65]: Shouldn't we update the language here as well?

File: packages/legend-query-builder/src/components/QueryUsageViewer.tsx [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+      (option) => option.title === selectedTab,
+    );
+
+    return (
+      <Dialog
+        open={resultState.isQueryUsageViewerOpened}
+        onClose={() => resultState.setIsQueryUsageViewerOpened(false)}
+      >
+        <Modal className="query-builder__usage-viewer__modal" darkMode={true}>
+          <ModalHeader title="Query Usage Options" />
```
Comment (akphi) [inline packages/legend-query-builder/src/components/QueryUsageViewer.tsx:60]:
```suggestion
          <ModalHeader title="Query Usage" />
```

**Review by akphi — CHANGES_REQUESTED** [review]

**Review by akphi — APPROVED** [review]

**Pull request merged**